targeted/curl_command_generator.js

```diff
@@ -0,0 +1,29 @@
+//it will generate and copy curl commands based on the request
+//released under the Apache v2.0 licence.
+//You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//auther:@haseebeqx
+
+function invokeWith(msg) {
+	string = "curl -i -s -k -X  '"+msg.getRequestHeader().getMethod().toString()+"'  \\\n";
+	header = msg.getRequestHeader().toString();
+	header = header.split('\n');
+	for(i=1;i<header.length;i++){
+		string += " -H '"+header[i].trim()+"' ";
+	}
+	string += " \\\n";
+	body = msg.getRequestBody().toString().trim();
```

Comment:
Since leading and trailing whitespace chars are removed from the string it changes its contents which most likely will also change its meaning. That can easily happen with "binary" bodies (for example, with AMF message, the ```trim()``` removes the version and header count (and possibly a couple more bytes, I think) which breaks the message).
The function ```addSlashes```might change the meaning of the request, too.

I'm OK with that, although it should be described in the script that not all types of messages are correctly handled, or that it's most appropriate for "text" bodies.